master/internal/db/postgres.go

```diff
@@ -1483,6 +1483,35 @@ WHERE id = :id`, setClause(toUpdate)), validation)
 	return nil
 }
 
+// AddModel adds the model to the database and sets its ID.
+func (db *PgDB) AddModel(m *model.Model) error {
+	var count int
+	err := db.namedGet(&count, `
+SELECT COUNT(*)
+FROM models
+WHERE name = :name`, m)
+	if err != nil {
+		return errors.Wrap(err, "error querying model table")
+	}
+	if count > 0 {
+		return errors.Errorf("duplicate model for name %s", m.Name)
+	}
+	err = db.namedGet(&m.ID, `
```

Comment:
non-blocking: return errors.Wrapf() instead of the if statement below?

Reply:
🤯 - wrapf returns nil if err is nil